File: codes/MaximumDifference2016/Solution.java
package codes.MaximumDifference2016;

public class Solution {
    public int maximumDifference(int[] nums) {
        int ans = -1;
        int numsLen = nums.length;
        int min = nums[0];
        for(int i = 0; i < numsLen; i++){
            if(nums[i]>min){
                ans = Math.max(ans, nums[i] - min);
            }
            min = Math.min(nums[i], min);
        }

        return ans;
    }
}

File: codes/MaximumDifference2016/solution.cs
public class Solution {
    public int MaximumDifference(int[] nums) {
        int res = -1, preMin = nums[0];
        for(int i = 1; i < nums.Length; ++ i){
            if(nums[i] > preMin){
                res = Math.Max(nums[i] - preMin, res);
            } else {
                preMin = nums[i];
            }
        }
        return res;
    }
}